### ISSUE TITLE
Golang Based Operator Tutorial: issues with the content of file memcached_controller.go

### ISSUE THREAD

**Issue by ErezArbell** [operator-framework/operator-sdk#4430]
Issues in page <https://sdk.operatorframework.io/docs/building-operators/golang/tutorial/>:

#### Issue 1
In the line:
```
For this example replace the generated controller file controllers/memcached_controller.go with the example memcached_controller.go implementation.
```
The link in `memcached_controller.go` need to be fixed.
With the current link (<https://github.com/operator-framework/operator-sdk/blob/v1.2.0/testdata/go/memcached-operator/controllers/memcached_controller.go>) you will get error in the test stage:
```
~/projects/memcached-operator$ make run ENABLE_WEBHOOKS=false
/home/erez/projects/memcached-operator/bin/controller-gen object:headerFile="hack/boilerplate.go.txt" paths="./..."
go fmt ./...
go vet ./...
# github.com/example-inc/memcached-operator/controllers
controllers/memcached_controller.go:185:11: cannot use r (type *MemcachedReconciler) as type reconcile.Reconciler in argument to controllerruntime.NewControllerManagedBy(mgr).For(&"github.com/example-inc/memcached-operator/api/v1alpha1".Memcached literal).Owns(&"k8s.io/api/apps/v1".Deployment literal).Complete:
        *MemcachedReconciler does not implement reconcile.Reconciler (wrong type for Reconcile method)
                have Reconcile(reconcile.Request) (reconcile.Result, error)
                want Reconcile(context.Context, reconcile.Request) (reconcile.Result, error)
# github.com/example-inc/memcached-operator/controllers
vet: controllers/memcached_controller.go:185:12: cannot use r (variable of type *MemcachedReconciler) as reconcile.Reconciler value in argument to ctrl.NewControllerManagedBy(mgr).For(&(cachev1alpha1.Memcached literal)).Owns(&(appsv1.Deployment literal)).Complete: wrong type for method Reconcile (have func(req sigs.k8s.io/controller-runtime/pkg/reconcile.Request) (sigs.k8s.io/controller-runtime/pkg/reconcile.Result, error), want func(context.Context, sigs.k8s.io/controller-runtime/pkg/reconcile.Request) (sigs.k8s.io/controller-runtime/pkg/reconcile.Result, error))
make: *** [Makefile:70: vet] Error 2
```
Should be <https://github.com/operator-framework/operator-sdk/blob/latest/testdata/go/v3/memcached-operator/controllers/memcached_controller.go>

#### Issue 2
In the content of this file (<https://github.com/operator-framework/operator-sdk/blob/latest/testdata/go/v3/memcached-operator/controllers/memcached_controller.go>), this:
```
cachev1alpha1 "github.com/example/memcached-operator/api/v1alpha1"
```
should be replaced with 
```
cachev1alpha1 "github.com/example-inc/memcached-operator/api/v1alpha1"
```

**Comment by varshaprasad96** [comment]
This was fixed in master (https://github.com/operator-framework/operator-sdk/pull/4393/files), but needs to back ported. Thanks for pointing this out, will cheery pick this PR for previous versions.

**Comment by varshaprasad96** [comment]
@ErezArbell This [PR](https://github.com/operator-framework/operator-sdk/pull/4465) fixes a few issues found in tutorial of 1.3.x. Please free to take a look and let us know if there is anything missing.
